libs/designsystem/shared/floating/src/floating.directive.ts

```diff
@@ -0,0 +1,368 @@
+import {
+  Directive,
+  ElementRef,
+  EventEmitter,
+  HostListener,
+  Input,
+  OnDestroy,
+  OnInit,
+  Output,
+  Renderer2,
+} from '@angular/core';
+import {
+  autoPlacement,
+  autoUpdate,
+  computePosition,
+  offset,
+  shift,
+  Strategy,
+} from '@floating-ui/dom';
+import { ComputePositionConfig, Middleware, Placement } from '@floating-ui/core/src/types';
+import { DesignTokenHelper } from '@kirbydesign/core';
+import { from } from 'rxjs';
+import { PortalDirective } from '@kirbydesign/designsystem/shared/portal';
+
+export type TriggerEvent = 'hover' | 'click' | 'focus';
+
+export enum FloatingOffset {
+  none = 0,
+  small = 5,
+  medium = 10,
+  large = 15,
+}
+
+interface EventMethods {
+  event: string;
+  method: () => void;
+}
+
+type EventListener = () => void;
```

Comment:
The Typescript dom lib has a similar type, so you should be able to remove this line.